### PR TITLE
supporting rolling aggregates

### PR DESCRIPTION
**Pull request by chungg**
this implements rolling aggregates which works similar to how
pandas rolling works but more trivial. this differs from deprecated
moving_average by correctly matching points to trailing timestamp
rather than leading timestamp.

it does not return the NaN values that result in shift in values.

#186

### CODE REVIEW

File: gnocchi/aggregates/rolling.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    # http://arogozhnikov.github.io/2015/09/30/NumpyTipsAndTricks2.html
+    stride = values.strides[0]
+    return as_strided(values, shape=[len(values) - window + 1, window],
+                      strides=[stride, stride])
+
+
+def build_array(storage, metric, start, stop, granularity):
+    serie = storage.get_measures(metric, from_timestamp=start,
+                                 to_timestamp=stop, granularity=granularity)
+    ts = np.array(serie, dtype=[('timestamps', '<datetime64[ns]'),
```
Comment (jd) [inline gnocchi/aggregates/rolling.py]: I'm feeling we could change `fetch` in Carbonara to directly return that format.

File: gnocchi/aggregates/rolling.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                                ('granularity', '<timedelta64[ns]'),
+                                ('values', '<d')])
+    return ts
+
+
+class RollingMean(aggregates.CustomAggregator):
+    def compute(self, storage_obj, metric, start, stop, granularity, window):
```
Comment (jd) [inline gnocchi/aggregates/rolling.py]: seems `@staticmethod`

**Comment by chungg** [comment]
not sure what you mean by old api?

this is a re-aggregation so it is slightly misleading as i've currently implemented it. i could switch this over to aggregation endpoint but that's generally used for multiple metrics...

**Comment by jd** [comment]
By old API I meant the fact this it is passed as an `aggregation` parameter. If that fits well with that computing, I'm good with it. Just questioning it since I don't think anyone ever used the moving average one.

**Comment by chungg** [comment]
i don't think they did either -- it doesn't give correct values typical of sliding/moving/rolling windows computations.

i'll try moving this to reaggregation since that is what it is.

**Pull request merged**